File: rabbitmq-nodejs/consumer.js
const amqp = require('amqplib');

async function main() {
    const connection = await amqp.connect('amqp://localhost');
    const channel = await connection.createChannel();

    const queueName = 'hello';

    await channel.assertQueue(queueName, { durable: true }); 
    console.log('[*] Waiting for messages. To exit, press CTRL+C');

    channel.consume(queueName, (msg) => {
        console.log([x] Received ${msg.content.toString()});
    }, { noAck: true });
}

main().catch(console.error);

File: rabbitmq-python/consumer.py
import pika

def callback(ch, method, properties, body):
    print(f"[x] Received {body}")

connection = pika.BlockingConnection(pika.ConnectionParameters('localhost'))
channel = connection.channel()

queue_name = 'hello'
channel.queue_declare(queue=queue_name, durable=True)  # Establece durable en True

print("[*] Waiting for messages. To exit, press CTRL+C")
channel.basic_consume(queue=queue_name, on_message_callback=callback, auto_ack=True)

channel.start_consuming()
